test/jdk/java/awt/Window/Grab/GrabTest.java

```diff
@@ -67,61 +86,55 @@ public void eventDispatched(AWTEvent e) {
                 }
             }, sun.awt.SunToolkit.GRAB_EVENT_MASK);
 
-        f = new Frame("Frame");
-        f.setBounds(0, 0, 300, 300);
-        f.addMouseListener(new MouseAdapter() {
+            f = new Frame("Frame");
+            f.setBounds(0, 0, 300, 300);
+            f.addMouseListener(new MouseAdapter() {
                 public void mousePressed(MouseEvent e) {
                     System.out.println(e);
                     framePressed = true;
                 }
             });
 
-        f1 = new Frame("OtherFrame");
-        f1.setBounds(700, 100, 300, 300);
+            f1 = new Frame("OtherFrame");
+            f1.setBounds(700, 100, 300, 300);
 
-        w = new Window(f);
-        w.setLayout(new FlowLayout());
-        b = new Button("Press");
-        b.addActionListener(new ActionListener() {
+            w = new Window(f);
+            w.setLayout(new FlowLayout());
+            b = new Button("Press");
+            b.addActionListener(new ActionListener() {
                 public void actionPerformed(ActionEvent e) {
                     System.out.println(e);
                     buttonPressed = true;
                 }
             });
-        w.add(b);
-        w.setBounds(400, 100, 300, 300);
-        w.setBackground(Color.blue);
-        w.addMouseListener(new MouseAdapter() {
+            w.add(b);
+            w.setBounds(400, 100, 300, 300);
+            w.setBackground(Color.blue);
+            w.addMouseListener(new MouseAdapter() {
                 public void mousePressed(MouseEvent e) {
                     System.out.println(e);
                     windowPressed = true;
                 }
             });
 
-        f.setVisible(true);
-        w.setVisible(true);
+            f.setVisible(true);
+            w.setVisible(true);
```

Comment:
```suggestion
            f.setVisible(true);
            w.setVisible(true);
            robot.delay(500);
```

Since it is unstable test, adding a delay here after the first set of frame & window are made visible might be good.

Reply:
+1, but it should be outside of the invokeAndWait. And there are other setVisible calls.